### PR TITLE
V0.1

### PR DESCRIPTION
**Pull request by akhil8044**
Updated Exceptions and added net value to price

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -50,8 +50,19 @@
 			<plugin>
 				<groupId>org.springframework.boot</groupId>
 				<artifactId>spring-boot-maven-plugin</artifactId>
+				<configuration>
+					<mainClass>DemoApplication</mainClass>
+					<layout>WAR</layout>
```
Comment (chandu5884) [inline pom.xml]: Why do you need a WAR layout?

File: src/main/java/com/demo/Model/Price.java [hunk 2]
```diff
@@ -6,5 +6,5 @@
 @Data
 @AllArgsConstructor
 public class Price {
-    Long totalPrice, totalGst, cGst,sGst;
```
Comment (chandu5884) [inline src/main/java/com/demo/Model/Price.java:9]: Can we use all these as seperate fields, please?

File: src/main/java/com/demo/controller/DemoApplicationController.java [hunk 3]
```diff
@@ -14,6 +14,8 @@
 @RequiredArgsConstructor
 public class DemoApplicationController {
 
+    //Updated
```
Comment (chandu5884) [inline src/main/java/com/demo/controller/DemoApplicationController.java]: Is this comment needed?

File: pom.xml [hunk 4]
```diff
@@ -50,8 +50,19 @@
 			<plugin>
 				<groupId>org.springframework.boot</groupId>
 				<artifactId>spring-boot-maven-plugin</artifactId>
```
Comment (chandu5884) [inline pom.xml]: Did we add the swagger dependency?

File: src/main/java/com/demo/controller/DemoApplicationController.java [hunk 5]
```diff
@@ -14,6 +14,8 @@
...
+
     private final PlansService plansService;
 
     @GetMapping(value = "/Demo/getPlans/{planType}")
```
Comment (chandu5884) [inline src/main/java/com/demo/controller/DemoApplicationController.java]: Can we add the swagger description of what this endpoint does?

File: src/main/java/com/demo/controller/DemoApplicationController.java [hunk 6]
```diff
@@ -14,6 +14,8 @@
 @RequiredArgsConstructor
```
Comment (chandu5884) [inline src/main/java/com/demo/controller/DemoApplicationController.java:19]: can we add a lombok `@SLf4j` annotation here for logging?

File: src/main/java/com/demo/controller/DemoApplicationControllerAdvice.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import com.demo.exceptions.ErrorResponse;
+import com.demo.exceptions.PlanExceptions;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+
+@ControllerAdvice
+public class DemoApplicationControllerAdvice extends ResponseEntityExceptionHandler {
```
Comment (chandu5884) [inline src/main/java/com/demo/controller/DemoApplicationControllerAdvice.java:15]: lombok logging annotation here as well

File: src/main/java/com/demo/dao/PlansDao.java [hunk 8]
```diff
@@ -3,19 +3,21 @@
 import com.demo.Model.Benefits;
 import com.demo.Model.Plans;
 import com.demo.Model.Price;
-import org.springframework.stereotype.Component;
```
Comment (chandu5884) [inline src/main/java/com/demo/dao/PlansDao.java:6]: can we please remove unwanted imports?

File: src/main/java/com/demo/service/PlansService.java [hunk 9]
```diff
@@ -14,6 +16,10 @@
     private final PlansDao plansDao;
 
     public List<Plans> getPlans(String planType) {
```
Comment (chandu5884) [inline src/main/java/com/demo/service/PlansService.java]: Can we use an enum for planType and use switch cases here? is that a possibility?

File: pom.xml (hunk 1 above)
Comment (akhil8044) [reply pom.xml]: Updated to Jar

File: src/main/java/com/demo/Model/Price.java (hunk 2 above)
Comment (akhil8044) [reply src/main/java/com/demo/Model/Price.java:9]: Updated as per request. Also change the type to Float.

File: src/main/java/com/demo/controller/DemoApplicationController.java (hunk 3 above)
Comment (akhil8044) [reply src/main/java/com/demo/controller/DemoApplicationController.java]: Removed

File: pom.xml (hunk 4 above)
Comment (akhil8044) [reply pom.xml]: Yes, I have integrated swagger now to the project

File: src/main/java/com/demo/controller/DemoApplicationController.java (hunk 5 above)
Comment (akhil8044) [reply src/main/java/com/demo/controller/DemoApplicationController.java]: Added swagger documentation for all the required endpoints and Models.

File: src/main/java/com/demo/dao/PlansDao.java (hunk 8 above)
Comment (akhil8044) [reply src/main/java/com/demo/dao/PlansDao.java:6]: done

File: src/main/java/com/demo/controller/DemoApplicationController.java (hunk 6 above)
Comment (akhil8044) [reply src/main/java/com/demo/controller/DemoApplicationController.java:19]: Added @slf4j annotation as requested

File: src/main/java/com/demo/controller/DemoApplicationControllerAdvice.java (hunk 7 above)
Comment (akhil8044) [reply src/main/java/com/demo/controller/DemoApplicationControllerAdvice.java:15]: Done

File: src/main/java/com/demo/service/PlansService.java (hunk 9 above)
Comment (akhil8044) [reply src/main/java/com/demo/service/PlansService.java]: Enum created for planType. But, switch cases cannot be compared with String for enum(Constant expression required error) . So, used enums and iteration for validation. Suggest if any changes.

**Pull request merged**